lib/order.rb

```diff
@@ -0,0 +1,17 @@
+class Order
+
+attr_accessor :ordered_dishes
+
+  def initialize
+    @ordered_dishes = { }
+  end
+
+  def total
+
+  end
+
+  def add_item(dish, quantity = 1)
+    added_item = {dish => @menu[dish]} #trying to access the hash in another class but not much luck with it. 
```

Comment:
RSpec gave the error:

```
NoMethodError:
       undefined method `[]' for nil:NilClass
```

So, Ruby can't use the method written with square brackets on a NilClass object. That is a hint that `@menu` is `nil` in this case. Why would a variable evaluate to nil?

<details><summary>Click here once you've answered the question</summary>
Right, if you haven't initialized a variable, it returns `nil`.

So menu has to be accessible within this class Order. Probably you'll pass in a Menu object as an argument when you create the Order object.
</details>